casper/src/test/resources/RegistryTest.rho

```diff
@@ -0,0 +1,77 @@
+//requires RhoSpec
+new
+  RhoSpecCh,
+  rl(`rho:registry:lookup`),
+  rs(`rho:registry:insertSigned:secp256k1`),
+  rr(`rho:registry:insertArbitrary`),
+  sign(`rho:test:crypto:secp256k1Sign`),
+  secpVerify(`rho:crypto:secp256k1Verify`),
+  blake2b256(`rho:crypto:blake2b256Hash`),
+  test_secp,
+  test_insertSignedAndLookup,
+  test_insertArbitraryAndLookup
+in {
+  rl!(`rho:id:zphjgsfy13h1k85isc8rtwtgt3t9zzt5pjd5ihykfmyapfc4wt3x5h`, *RhoSpecCh) |
+  for(@(_, RhoSpec) <- RhoSpecCh) {
+    @RhoSpec!("testSuite",
+      [
+        ("Verifying a signature should work", *test_secp),
+        ("InsertSigned and lookup should work", *test_insertSignedAndLookup),
+        ("InsertArbitrary and lookup should work", *test_insertArbitraryAndLookup)
+      ])
+  } |
+
+  contract test_secp(rhoSpec, _, ackCh) = {
+    match ("04cc94ab15247b0db1b8219388218eb7461ac74b3d88d4da8c1816fac8e258f5e0f5db9db6bb82b0cc066589dfe77a0d7449db295dab248fb93855ba91813154a9".hexToBytes(),
+           "4d24bf7b37a4b3dc417b8e0e03a0f232b31a535f4880a73575ac855acae49640".hexToBytes()) {
+      (pk, sk) => {
+        new signCh, verifyCh, retCh, hashCh in {
+          blake2b256!("foo", *hashCh) |
+          for (@hash <- hashCh) {
+            sign!(hash, sk, *signCh) |
+            for (@sig <- signCh) {
+              secpVerify!(hash, sig, pk, *verifyCh) |
+              for (@b <- verifyCh) {
+                retCh!(b) |
+                rhoSpec!("assert", (true, "== <-", *retCh), "Expecting true", *ackCh)
```

Comment:
You can skip the verifyCh -> retCh sending, and pass verifyCh to the assertion direclty

Reply:
Done